BUILDER_DEV.md

```diff
@@ -1,390 +1,294 @@
+* Access to the Habitat 1Password account (NOTE - we are working on a solution to make this not required, but it is necessary for the time being)
+* All of the following steps should be run within your Virtual Machine
+* A [Github Personal Access token](https://help.github.com/articles/creating-a-personal-access-token-for-the-command-line/) with all repo and all user permissions.
+* The [Habitat Builder App](https://github.com/apps/habitat-builder) installed on your Github account.
+* Open your Sudoers file at /etc/sudoers and remove the "secure_path" section (this is required for the make scripts to run correctly)
+
+```
+$ sudo visudo -f /etc/sudoers
+```
+
+### Builder Setup
+* Update your system
+```
+$ sudo apt-get update
+```
+
+* Install git
+```
+$ sudo apt-get install git
+```
+
+* Install curl
+```
+$ sudo apt-get install curl
+```
+
+* Clone the [Habitat repo](https://github.com/habitat-sh/habitat)
+```
+$ git clone https://github.com/habitat-sh/habitat.git
+```
+
+* Install the [Habitat CLI](https://www.habitat.sh/docs/get-habitat/)
+* Copy habitat-builder-dev.2017-10-02.private-key.pem from 1Password into path/to/habitat/repo/.secrets/builder-github-app.pem (NOTE - this is required as of 12/11/17 - we are working on a solution to make access to the Habitat 1Password vault not required for dev environment setup).
+* Set up some environmental variables (I put these in my .bashrc)
+
+```
+export HAB_AUTH_TOKEN="<your github token>"
+export HAB_BLDR_URL="http://localhost:9636"
+export HAB_ORIGIN="core"
+```
+
+* symlink /src to path/to/your/habitat/repo
+
+```
+$ sudo ln -s path/to/your/habitat/repo /src
 ```
 
-`config_jobsrv.toml`
-```toml
-key_dir = "/path/to/bldr-key-pair"
+* Cd into your Habitat repo
 
-[archive]
-backend = "local"
-local_dir = "/tmp"
 ```
+$ cd path/to/your/habitat/repo
+```
+
+* Run the provision script
+
+```
+$ sudo ./support/linux/provision.sh
+```
+
+* After the script completes, source cargo/rust
+```
+$ source $HOME/.cargo/env
+```
+
+* Run the build-srv makefile (NOTE - if you receive an error about cargo not being found, make sure you have removed the "secure_path" section of /etc/sudoers)
 
-`config_sessionsrv.toml`
-```toml
-[permissions]
-admin_team = 1995301
-build_worker_teams = [2555389]
-early_access_teams = [1995301]
 ```
+$ sudo make build-srv
+```
+
+* Open up the builder-worker file
 
-`config_worker.toml`
-```toml
+```
+$ sudo vim /hab/svc/builder-worker
+```
+
+* Add your Github Auth Key:
+
+**/hab/svc/builder-worker/config.toml**
+```
+auth_token = ""
+```
+
+Should become:
+
+**/hab/svc/builder-worker/config.toml**
+```
 auth_token = "<your github token>"
-bldr_url = "http://localhost:9636"
-auto_publish = true
-airlock_enabled = false
 ```
 
+* Run this command (it may take awhile) to start all the Builder services
+
+```
+$ sudo -E make bldr-run
```

Comment:
This can be `bldr-run-no-build` since we just did the build step.

Reply:
We should also add a comment to make sure the services all start properly - wait till the end when you can see the worker heartbeat debug entries in the log.  Also as a troubleshooting step, if the services don't start because some were already running from a previous attempt, there is now a 'make bldr-kill' command that may be useful.